processor/transform/variables.ts

```diff
@@ -12,8 +12,21 @@ const variables =
     visit(tree, (node, index, parent) => {
       if (!['mdxFlowExpression', 'mdxTextExpression'].includes(node.type) || !('value' in node)) return;
 
-      const match = node.value.match(/^user\.(?<value>.*)$/);
-      if (!match) return;
+      // @ts-expect-error - estree is not defined on our mdx types?!
+      if (node.data.estree.type !== 'Program') return;
+      // @ts-expect-error - estree is not defined on our mdx types?!
+      const [expression] = node.data.estree.body;
+      if (
+        !expression ||
+        expression.type !== 'ExpressionStatement' ||
+        !['Literal', 'Identifier'].includes(expression.expression.property?.type)
+      )
+        return;
+
+      const name =
+        expression.expression.property.type === 'Identifier'
+          ? expression.expression.property.name
+          : expression.expression.property.value;
```

Comment:
I like this a lot more than using MDX, but it seems like this new logic will apply to _all_ `{vars}` (not just `user.` prefixed ones…) Does that matter at all?

Reply:
OH shit, I forget to check the module is named `user`. Good catch!